lib/commons/aria/get-role.js

```diff
@@ -22,17 +106,37 @@ import AbstractVirtuaNode from '../../core/base/virtual-node/abstract-virtual-no
 function getRole(node, { noImplicit, fallback, abstracts, dpub } = {}) {
 	const vNode =
 		node instanceof AbstractVirtuaNode ? node : getNodeFromTree(node);
+	node = vNode.actualNode;
 	if (vNode.props.nodeType !== 1) {
 		return null;
 	}
 	const explicitRole = getExplicitRole(vNode, { fallback, abstracts, dpub });
 
-	// Get the implicit role, if permitted
-	if (!explicitRole && !noImplicit) {
-		return getImplicitRole(vNode);
+	if (explicitRole && !['presentation', 'none'].includes(explicitRole)) {
+		return explicitRole;
+	}
+
+	// role conflict resolution
+	// note: Chrome returns a list with resolved role as "generic"
+	// instead of as a list
+	// (e.g. <ul role="none" aria-label><li>hello</li></ul>)
+	// we will return it as a list as that is the best option.
+	// Source: https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none
+	// See also: https://github.com/w3c/aria/issues/1270
+	const hasGlobalAria = lookupTable.globalAttributes.find(attr =>
+		vNode.hasAttr(attr)
+	);
+	if (hasGlobalAria || isFocusable(node)) {
+		// return null if there is a conflict resolution but no implicit
+		// has been set as the explicit role is not the true role
+		return noImplicit ? null : resolveImplicitRole(vNode);
+	}
+
+	if (!noImplicit && !explicitRole) {
+		return resolveImplicitRole(vNode);
```

Comment:
Just for the sake of readability
```suggestion
	if (noImplicit && !explicitRole) {
		return null;
```